light_controlled_intersection_tactical_plugin/src/light_controlled_intersection_tactical_plugin.cpp

```diff
@@ -180,10 +155,51 @@ namespace light_controlled_intersection_tactical_plugin
             && rclcpp::Time(last_trajectory_.trajectory_points.back().target_time).seconds() > rclcpp::Time(req->header.stamp).seconds())
         {
             resp->trajectory_plan = last_trajectory_;
-            RCLCPP_DEBUG_STREAM(rclcpp::get_logger("light_controlled_intersection_tactical_plugin"), "Last Traj's target time: " << rclcpp::Time(last_trajectory_.trajectory_points.back().target_time).seconds() << ", and stamp:" << rclcpp::Time(req->header.stamp).seconds() << ", and scheduled: " << std::to_string(last_successful_scheduled_entry_time_));
+            RCLCPP_DEBUG_STREAM(rclcpp::get_logger("light_controlled_intersection_tactical_plugin"), "Last Traj's target time: " << std::to_string(rclcpp::Time(last_trajectory_.trajectory_points.back().target_time).seconds()) << ", and stamp:" << std::to_string(rclcpp::Time(req->header.stamp).seconds()) << ", and scheduled: " << std::to_string(last_successful_scheduled_entry_time_));
             RCLCPP_DEBUG_STREAM(rclcpp::get_logger("light_controlled_intersection_tactical_plugin"), "EDGE CASE: USING LAST TRAJ: " << (int)last_case_.get());
         }  
-        else if (trajectory.trajectory_points.size () < 2)
+
+        if (!resp->trajectory_plan.trajectory_points.empty()) // if has valid trajectory saved from before return
```

Comment:
I'm not seeing a check to validate that this trajectory actually makes sense. 
Before just defaulting to the last trajectory there should be some form of validation that the vehicle is still following it within some error threshold.

Reply:
We discussed offline and we want to compare more in position and time than speed. Since that's what is relevant for crossing the stop bar.
"Am I within 3m of my expected position at the current time"
As far as the current changes go, we are tabling further implementation as enhancement and merging this version. For the record, the vehicle is preventing RED light entrance by following:

All decisions to stop the vehicle from going through red light happens on strategic level. Since it involves a different plugin, it is guaranteed to follow that stopping trajectory.
If trajectory smoothing algorithm doesnt have valid solution it decides to cruise and stop at red.
Furthermore, I think this might be irrelevant to our specific discussion, but on a strategic level, there is basically only one edge case we identified that lets the vehicle go through red but we discussed it with Ying Yang today that more implementation should be held off until we have resources to do so.
On a tactical level, there is no decisions being made using lights. Only decisions are whether if
new TScase is same as before and last traj has more than 2 points and not within its last 1 sec trajectory, then use last trajectory
if not all above, use new trajectory

Reply:
Created new enhancement issue: 
https://github.com/usdot-fhwa-stol/carma-platform/issues/2006